protocol/payment.go

```diff
@@ -72,147 +67,86 @@ type PaymentRequestMessageBody struct {
 
 // PaymentRequestInfo represents the payments request information.
 type PaymentRequestInfo struct {
-	Type        string                          `json:"type,omitempty"`
 	Credentials []PaymentRequestInfoCredentials `json:"credentials"`
 	Description string                          `json:"description"`
 	Data        PaymentRequestInfoData          `json:"data"`
```

Comment:
Could we just put []PaymentRequestInfoDataItem here for Data field ? What the reason to wrap in type ?